src/Makefile

```diff
@@ -20,4 +20,4 @@ fst-iterators.o: jopenfst.h fst-iterators.cc
 
 jopenfst.so: weight.o fst-class.o fst-ops.o fst-iterators.o
 	g++ -o jopenfst.so -fPIC -shared weight.o fst-class.o fst-ops.o \
-               fst-iterators.o $(LIB)/libfst.so $(LIB)/libfstscript.so
+               fst-iterators.o -Wl,-rpath=$(LIB) $(LIB)/libfst.so $(LIB)/libfstscript.so
```

Comment:
this also will need to get changed on apple -- I suggest just trying 
```
-Wl,-rpath=$(LIB) -lfstscript  -lfst 
```
it should chose their own affixes

Reply:
You might need to add -L$(LIB):
```
-Wl,-rpath=$(LIB) -L$(LIB) -lfstscript  -lfst 
```